### PR TITLE
Reenable dotty-compiler-plugin test

### PR DESCRIPTION
**Pull request by smarter**
https://github.com/lampepfl/dotty/issues/5671 is fixed now.

### CODE REVIEW

**Comment by eed3si9n** [comment]
Travis CI says:

```
[info] [info] Compiling 1 Scala source to /tmp/sbt_e5f965f1/plugin/target/scala-0.14/classes ...
[info] [error] /tmp/sbt_e5f965f1/plugin/DivideZero.scala:2:14: value tools is not a member of dotty
[info] [error] import dotty.tools.dotc._
[info] [error]              ^
[info] [error] /tmp/sbt_e5f965f1/plugin/DivideZero.scala:23:30: <error value tools is not a member of dotty>#dotc.List does not take type parameters
[info] [error]   override def init(options: List[String]): List[PluginPhase] = this :: Nil
[info] [error]                              ^
[info] [error] /tmp/sbt_e5f965f1/plugin/DivideZero.scala:23:45: <error value tools is not a member of dotty>#dotc.List does not take type parameters
[info] [error]   override def init(options: List[String]): List[PluginPhase] = this :: Nil
[info] [error]                                             ^
[info] [error] /tmp/sbt_e5f965f1/plugin/DivideZero.scala:33:63: Stable identifier required, but 31m35m<error value tools is not a member of dotty>31m0m#31m35mdotc31m0m.Nil found
[info] [error]     case tpd.Apply(fun, tpd.Literal(Constants.Constant(v)) :: Nil) if isNumericDivide(fun.symbol) && v == 0 =>
[info] [error]                                                               ^
[info] [error] /tmp/sbt_e5f965f1/plugin/DivideZero.scala:33:91: value symbol is not a member of Any
[info] [error]     case tpd.Apply(fun, tpd.Literal(Constants.Constant(v)) :: Nil) if isNumericDivide(fun.symbol) && v == 0 =>
[info] [error]                                                                                           ^
[info] [error] 5 errors found
[info] [error] (plugin / Compile / compileIncremental) Compilation failed
```

**Comment by eed3si9n** [comment]
We're now getting ordering conflict for the phase. Looks like a progress.

```
[info] [info] Compiling 1 Scala source to /tmp/sbt_762bcd99/target/scala-0.14/classes ...
[info] [info] exception occurred while compiling /tmp/sbt_762bcd99/src/main/scala/hello/Hello.scala
[info] java.lang.Exception: Ordering conflict for phase divideZero while compiling /tmp/sbt_762bcd99/src/main/scala/hello/Hello.scala
[info] [error] ## Exception when compiling 1 sources to /tmp/sbt_762bcd99/target/scala-0.14/classes
[info] [error] java.lang.Exception: Ordering conflict for phase divideZero
[info] [error] dotty.tools.dotc.plugins.Plugins$.schedule$$anonfun$11$$anonfun$2(Plugins.scala:267)
```

**Comment by smarter** [comment]
> [info] java.lang.Exception: Ordering conflict for phase divideZero while compiling /tmp/sbt_762bcd99/src/main/scala/hello/Hello.scala

@liufengyun Do you know why this might be happening ?

**Comment by liufengyun** [comment]
@smarter I think the problem is with the phase `LinkAll`, which has been removed.

Remove the following line will work:

https://github.com/sbt/sbt/blob/develop/sbt/src/sbt-test/compiler-project/dotty-compiler-plugin/plugin/DivideZero.scala#L22

I wondered why the plugin code compiled.

**Comment by smarter** [comment]
This PR already updates this line: https://github.com/sbt/sbt/pull/4641/files#diff-ddb6ec378e5551c95e253f5a1a088889R23

It compiled previously because the test was using an old version of dotty.

File: sbt/src/sbt-test/compiler-project/dotty-compiler-plugin/plugin/DivideZero.scala [hunk 1]
```diff
@@ -10,31 +10,32 @@ import transform.MegaPhase.MiniPhase
...
+/** Compiler plugin that emits an error when compiling a division by zero */
 class DivideZero extends PluginPhase with StandardPlugin {
   val name: String = "divideZero"
   override val description: String = "divide zero check"
 
   val phaseName = name
 
   override val runsAfter = Set(Pickler.name)
-  override val runsBefore = Set(LinkAll.name)
+  override val runsBefore = Set(Staging.name)
```
Comment (liufengyun) [inline sbt/src/sbt-test/compiler-project/dotty-compiler-plugin/plugin/DivideZero.scala]: `Staging` is before `Pickler` -- so we need to change the ordering.

Comment (smarter) [reply sbt/src/sbt-test/compiler-project/dotty-compiler-plugin/plugin/DivideZero.scala]: I copy-pasted that file from https://github.com/lampepfl/dotty/blob/d78b8c0c1b34ecbbd052acd41e911daced1a95a1/sbt-dotty/sbt-test/sbt-dotty/compiler-plugin/plugin/DivideZero.scala#L23 so something must be wrong with the way we run our tests then :)

**Pull request merged**